### PR TITLE
Smoke test: preliminary sqlite tests

### PR DESCRIPTION
**Pull request by tldahlgren**
This PR adds install tests consisting of a version check and a series of calls on a pre-built database.

It has been tested against the following `sqlite` versions: `3.26.0`, `3.27.0`, `3.28.0`, `3.29.0`, `3.30.0`, and `3.31.1`.

### CODE REVIEW

File: lib/spack/spack/package.py [hunk 1]
```diff
@@ -1668,6 +1668,7 @@ def run_test(self, exe, options=[], expected=[], status=None,
     def _run_test_helper(self, runner, options, expected, status, installed,
                          purpose):
         status = [status] if not isinstance(status, list) else status
+        expected = [expected] if not isinstance(expected, list) else expected
```
Comment (becker33) [inline lib/spack/spack/package.py]:
```suggestion
        expected = [expected] if isinstance(expected, six.string_types) else expected
```

That way we can use generators and other iterables seamlessly in place of a list.

**Review by becker33 — CHANGES_REQUESTED** [review]
The sqlite portion looks great. One request on the run_test portion

File: lib/spack/spack/package.py (hunk 1 above)
Comment (tldahlgren) [reply lib/spack/spack/package.py:1671]: How are six.string_types not covered by the original code?

Comment (tldahlgren) [reply lib/spack/spack/package.py]: Will also change the other list-based arguments for consistency.

**Comment by tldahlgren** [comment]
I preserved the changes in this PR in two separate commits so they can be merged without squashing.

File: lib/spack/spack/package.py (hunk 1 above)
Comment (tldahlgren) [reply lib/spack/spack/package.py]: Done.

**Pull request merged**